tools/ci_build/github/android/build_aar_package.py

```diff
@@ -0,0 +1,166 @@
+
+    if 'android_api' in _build_settings_data:
+        build_params += ['--android_api=' + str(_build_settings_data['android_api'])]
+    else:
+        build_params += ['--android_api=' + str(DEFAULT_ANDROID_API)]
+
+    build_settings['build_params'] = build_params
+    return build_settings
+
+
+def _build_aar(args):
+    build_settings = _parse_build_settings(args)
+    build_dir = args.build_dir
+
+    # Setup temp environment for building
+    my_env = os.environ.copy()
+    my_env['ANDROID_HOME'] = build_settings['android_sdk_path']
+    my_env['ANDROID_NDK_HOME'] = build_settings['android_ndk_path']
+
+    # Temp dirs to hold building results
+    _intermediates_dir = os.path.join(build_dir, 'intermediates')
+    _build_flavor = build_settings['build_flavor']
+    _aar_dir = os.path.join(_intermediates_dir, 'aar', _build_flavor)
+    _jnilibs_dir = os.path.join(_intermediates_dir, 'jnilibs', _build_flavor)
+    _base_build_command = [
+        'python3', BUILD_PY, '--config=' + _build_flavor
+    ] + build_settings['build_params']
+
+    # Build binary for each ABI, one by one
+    for abi in build_settings['build_abis']:
+        _build_dir = os.path.join(_intermediates_dir, abi)
+        _build_command = _base_build_command + [
+            '--android_abi=' + abi,
+            '--build_dir=' + _build_dir
+        ]
+
+        if args.include_ops_by_config is not None:
+            _build_command += ['--include_ops_by_config=' + args.include_ops_by_config]
+
+        subprocess.run(_build_command, env=my_env, shell=False, check=True, cwd=REPO_DIR)
+
+        # create symbolic links for libonnxruntime.so and libonnxruntime4j_jni.so
+        # to jnilibs/[abi] for later compiling the aar package
+        _jnilibs_abi_dir = os.path.join(_jnilibs_dir, abi)
+        os.makedirs(_jnilibs_abi_dir, exist_ok=True)
+        for lib_name in ['libonnxruntime.so', 'libonnxruntime4j_jni.so']:
+            _target_lib_name = os.path.join(_jnilibs_abi_dir, lib_name)
+            # if the symbolic already exists, delete it first
+            if os.path.exists(_target_lib_name):
+                os.remove(_target_lib_name)
+            os.symlink(os.path.join(_build_dir, _build_flavor, lib_name), _target_lib_name)
+
+    # The directory to publish final AAR
+    _aar_publish_dir = os.path.join(build_dir, 'aar_out', _build_flavor)
+    os.makedirs(_aar_publish_dir, exist_ok=True)
+
+    # get the common gradle command args
+    _gradle_command = [
+        'gradle',
+        '--no-daemon',
+        '-b=build-android.gradle',
+        '-c=settings-android.gradle',
+        '-DjniLibsDir=' + _jnilibs_dir,
+        '-DbuildDir=' + _aar_dir,
+        '-DpublishDir=' + _aar_publish_dir
+    ]
+
+    # clean, build, and publish to a local directory
+    subprocess.run(_gradle_command + ['clean'], env=my_env, shell=False, check=True, cwd=JAVA_ROOT)
+    subprocess.run(_gradle_command + ['build'], env=my_env, shell=False, check=True, cwd=JAVA_ROOT)
```

Comment:
Would there be anything to clean, or is this just being safe?

Reply:
This is for local custom build, so you may try a couple times, it's better to clean it, for pipeline build there is not really anything to clean